adr/0010-release-via-tags.md

```diff
@@ -0,0 +1,27 @@
+# Release via tags
+
+Date: 2018-12-03
+
+## Status
+
+Proposed
+
+## Context
+
+eLife has some projects that are published as packages, e.g. for Python projects on pypi.
+
+Specific details may vary from project to project. For example one project may just pushed to pypi, another to Docker Hub, or even both. Some artefacts may just be pushed for releases, others on commit.
+
+For any publicly available artifact it's important to specify a version that can be traced back to the source control repository.
+
+## Decision
+
+A new release will be indicated by a new Git tag.
+
+Jenkins will trigger on a new tag to, beyond building and running tests, also push the release out as needed (e.g. pypi, Docker Hub).
+
+## Consequences
+
+Version numbers cannot be specified in source code, as the tag is the source of truth for them.
```

Comment:
Perhaps not specified... but you could certainly refer to old version numbers to keep behaviour consistent with older versions, which I see as the main reason for using version numbers in source.

Reply:
git tags can be deleted and reused. For example you could have the tag 'latest' that always points to the most recent passing build. They're not immutable like other aspects of git

Reply:
No, branches have this behavior; tags are supposed to be immutable and it's very complicated to [overwrite them](https://git-scm.com/docs/git-tag#_on_re_tagging) in clients that already have the tag.

I'll rephrase to make clear that the code can refer to version numbers in general, but shouldn't contain its own current version number.

Reply:
this link is also helpful:
https://git-scm.com/book/en/v2/Git-Basics-Tagging